### PR TITLE
Adapt username color to the app theme

### PR DESCRIPTION
**Pull request by devalexx**
As a user I have difficulties with reading some usernames in the chat. For example, light yellow in the white theme or black in the black theme.

Please, let me know what you think. And what about having it ON by default?

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -624,4 +624,5 @@
     <string name="move_to_shared_storage">Move to shared storage</string>
     <string name="move_to_app_storage">Move to app storage</string>
     <string name="download_moving">Moving</string>
+    <string name="theme_adapted_username_color">Adapt username color to the app theme</string>
```
Comment (crackededed) [inline app/src/main/res/values/strings.xml]: should we call this "Readable colors"? it should work with all messages now

File: app/src/main/res/xml/root_preferences.xml [hunk 2]
```diff
@@ -183,6 +183,13 @@
         app:iconSpaceReserved="false"
         app:singleLineTitle="false" />
 
+    <SwitchPreferenceCompat
+        android:defaultValue="false"
```
Comment (crackededed) [inline app/src/main/res/xml/root_preferences.xml]:
```suggestion
        android:defaultValue="true"
```

File: app/src/main/java/com/github/andreyasadchy/xtra/ui/view/chat/ChatView.kt [hunk 3]
```diff
@@ -95,6 +95,8 @@ class ChatView : ConstraintLayout {
                 emoteSize = context.convertDpToPixels(29.5f),
                 badgeSize = context.convertDpToPixels(18.5f),
                 randomColor = context.prefs().getBoolean(C.CHAT_RANDOMCOLOR, true),
+                isLightTheme = context.isLightTheme,
+                useThemeAdaptedUsernameColor = context.prefs().getBoolean(C.CHAT_THEME_ADAPTED_USERNAME_COLOR, false),
```
Comment (crackededed) [inline app/src/main/java/com/github/andreyasadchy/xtra/ui/view/chat/ChatView.kt]:
```suggestion
                useThemeAdaptedUsernameColor = context.prefs().getBoolean(C.CHAT_THEME_ADAPTED_USERNAME_COLOR, true),
```

File: app/src/main/res/values/strings.xml (hunk 1 above)
Comment (devalexx) [reply app/src/main/res/values/strings.xml]: yeah, makes sense, I've just added "based on the app theme" to make it obvious

**Pull request merged**